test/gtest/error_handling.cpp

```diff
@@ -120,7 +120,7 @@ class TestErrorHandling : public testing::TestWithParam<std::string> {
     void exchangeMetaData();
     nixlXferReqH* postXfer(enum nixl_xfer_op_t op, bool target_failure);
 
-    std::pair<bool, std::string> m_plugin_dir_backup = {false, ""};
+    std::vector<ScopedEnvVar>    m_env_vars;
```

Comment:
WDYT of abstracting even more?
vector+emplace_back is an implementation detail. Encapsulate it in an object, i.e `SetEnvVarContext m_context`.
Then in test ctor:
`m_context.push(key, value)` or something like that?